subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyPluginNotationConverter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2009 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.notations;
+
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.artifacts.ModuleIdentifier;
+import org.gradle.api.artifacts.MutableVersionConstraint;
+import org.gradle.api.internal.artifacts.DefaultModuleIdentifier;
+import org.gradle.api.internal.artifacts.dependencies.DefaultMinimalDependency;
+import org.gradle.api.internal.artifacts.dependencies.DefaultMutableVersionConstraint;
+import org.gradle.internal.exceptions.DiagnosticsVisitor;
+import org.gradle.internal.typeconversion.NotationConvertResult;
+import org.gradle.internal.typeconversion.NotationConverter;
+import org.gradle.internal.typeconversion.TypeConversionException;
+import org.gradle.plugin.use.PluginDependency;
+
+public class DependencyPluginNotationConverter implements NotationConverter<PluginDependency, MinimalExternalModuleDependency> {
+    private final NotationConverter<MinimalExternalModuleDependency, MinimalExternalModuleDependency> minimalConverter;
+
+    public DependencyPluginNotationConverter(NotationConverter<MinimalExternalModuleDependency, MinimalExternalModuleDependency> minimalConverter) {
+        this.minimalConverter = minimalConverter;
+    }
+
+    @Override
+    public void describe(DiagnosticsVisitor visitor) {
+        visitor.candidate("PluginDependency").example("plugin('org.gradle.java:1.0')");
+    }
+
+    @Override
+    public void convert(PluginDependency notation, NotationConvertResult<? super MinimalExternalModuleDependency> result) throws TypeConversionException {
+        ModuleIdentifier module = DefaultModuleIdentifier.newId(notation.getPluginId(), notation.getPluginId() + ".gradle.plugin");
```

Comment:
:x: This duplicates what is done in `PluginDependencyMarkerCoordinates.pluginNotation()`.

BTW, do we still need this `DependencyPluginNotationConverter`?

Reply:
That's right, but we need to pass `group` and `name` as separate arguments here, so we can't use `PluginDependencyMarkerCoordinates.pluginNotation()`. `DependencyPluginNotationConverter` is used in [DependencyNotationParser::create](https://github.com/gradle/gradle/pull/23685/files#diff-71fff22006de064962b3898877eca4046b5f8450f116a5245242123413cab16aR77), I don't think we can drop it.

Reply:
Ok. I still think we should move the mapping to plugin coordinates in a method in `PluginDependencyMarkerCoordinates`.

Reply:
Good point, I'll do that.